dataplane/src/packet_meta.rs

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+use routing::vrf::VrfId;
+use std::collections::HashMap;
+
+#[derive(Debug, Default)]
+pub struct InterfaceId(u32);
+#[allow(unused)]
+impl InterfaceId {
+    pub fn get_id(&self) -> u32 {
+        self.0
+    }
+    pub fn set_id(&mut self, id: u32) {
+        self.0 = id
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct BridgeDomain(u32);
+#[allow(unused)]
+impl BridgeDomain {
+    pub fn get_id(&self) -> u32 {
+        self.0
+    }
+    pub fn with_id(id: u32) -> Self {
+        Self(id)
+    }
+}
+
+#[allow(unused)]
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub enum DropReason {
```

Comment:
Do we want one global `DropReason` enum?  Isn't this a per network function construct?

For example, would we extend this type with more errors for IPsec?  Then more for Wireguard?  Then more for ...

I need to think about this.

On big error type seems to somewhat undermine the utility of exhaustive pattern matching

Reply:
Happy to have some hierarchy. I'd use a single type, though. You could have some hierarchy by nesting enums and have a category Ipsec, another for routing, etc.. But we have to start by something.